### ISSUE TITLE
Using Ember.inject.service in authenticator causes restore to not be called

### ISSUE THREAD

**Issue by fotinakis** [simplabs/ember-simple-auth#595]
For some reason in 0.8.0 (untested in other versions), using [Ember.inject.service](http://emberjs.com/api/classes/Ember.inject.html#method_service) in an Authenticator causes the `restore()` method to never be called.

For example:

```javascript
import Ember from 'ember';
import BaseAuthenticator from 'simple-auth/authenticators/base';

export default BaseAuthenticator.extend({
  store: Ember.inject.service(),

  restore: function() {
    debugger // BUG: we will never get here
  },
...
```

Originally I thought it was because of an authenticator internal conflict with the property being named `store`, but this also didn't work:
```javascript
emberStore: Ember.inject.service('store'),
```

**Comment by fotinakis** [comment]
It looks like this might be because the ember-data "service:store" is injected by a new "instance initializer" (which is run after normal initializers) and ember-simple-auth has not been updated to use the new initializer pattern (it also throws deprecation warnings).

**Comment by fotinakis** [comment]
It looks like this will be fixed by #522 -- thanks!
